packages/shared/src/mock/metadata-utils.ts

```diff
@@ -290,19 +290,22 @@ export function makeNxDataGroup<
     axesAttr,
     auxiliary = {},
     auxAttr,
+    attributes = [],
+    children = [],
     ...groupOpts
   } = opts;
 
   return makeNxGroup(name, 'NXdata', {
     ...groupOpts,
     attributes: [
-      ...(groupOpts.attributes ?? []),
+      ...attributes,
       makeStrAttr('signal', signal.name),
       ...(axesAttr ? [makeNxAxesAttr(axesAttr)] : []),
       ...(silxStyle ? [makeSilxStyleAttr(silxStyle)] : []),
       ...(auxAttr ? [makeNxAuxAttr(auxAttr)] : []),
     ],
     children: [
+      ...children,
```

Comment:
Doesn't really matter, but `...children` and `...attributes` should perhaps be last in the array. There's no risk of overriding anything since we're not destructuring objects, and it feels like custom attributes/children should come last.